### ISSUE TITLE
UMD and npm

### ISSUE THREAD

**Issue by klaemo** [ccampbell/mousetrap#192]
Would you accept a pull request that adds a [UMD](https://github.com/umdjs/umd/blob/master/returnExports.js) shim, so mousetrap can work in AMD and commonjs environments (and of course also as a browser global)?

Also it could you publish mousetrap to npm, so that we can deprecate [this shim](https://github.com/component/mousetrap)?

Would make me as a browserify user very happy.

(as per https://github.com/component/mousetrap/pull/1)

Thanks

**Comment by maxkueng** [comment]
I came here because of the same problem. Mousetrap [is on npm](https://www.npmjs.com/package/mousetrap) and it points to this repository. Sadly, it's not CommonJS compatible. I see pull request #145 adds an additional check for `module.exports` but wouldn't it be better to write the module in CommonJS and then make it AMD and *global* compatible with Browserify's `--standalone` option?

Like this:

```sh
browserify mousetrap.js --standalone Mousetrap
```

In [VictorJS](https://www.npmjs.com/package/victor) I make a build like this:

```sh
(echo '/*!' ; cat ./LICENSE ; echo '*/' ; browserify index.js --standalone Victor) > build/victor.js \
&& cat build/victor.js | uglifyjs -mc --comments '/^!/' > build/victor.min.js
```

It creates an UMD version - both minified and unminified -  for Bower and others, and adds the license at the top. And keeps the CommonJS version for Node and Browserify.

**Comment by mightyiam** [comment]
@maxkueng this project [seems inactive](https://github.com/ccampbell/mousetrap/commits/master).

I recommend [my spork](https://github.com/mightyiam/combokeys), which has already seen some community participation and has more than 400 stars.

**Comment by maxkueng** [comment]
@mightyiam yeah I noticed. I'm using your fork right now (thanks, btw!). But this project is still being used my a lot of people according to its download stats on npm. Would be nice it if could be somehow revived.

**Comment by donkeybonks** [comment]
@ccampbell the plugins are not exposed as common js modules.
